File: code-review-graph/tests/fixtures/sample.rb
require 'json'

module Auth
  class User
    attr_accessor :id, :name, :email

    def initialize(id, name, email)
      @id = id
      @name = name
      @email = email
    end

    def to_s
      "User(#{@id}, #{@name})"
    end
  end

  class UserRepository
    def initialize
      @users = {}
    end

    def find_by_id(id)
      @users[id]
    end

    def save(user)
      @users[user.id] = user
      puts "Saved #{user}"
    end

    def create_user(name, email)
      user = User.new(@users.size + 1, name, email)
      save(user)
      user
    end
  end
end

File: code-review-graph/tests/fixtures/sample.scala
package com.example.auth

import scala.collection.mutable
import scala.collection.mutable.{HashMap, ListBuffer}
import scala.util.Try
import scala.concurrent._

trait Repository[T]:
  def findById(id: Int): Option[T]
  def save(entity: T): Unit

case class User(id: Int, name: String, email: String)

class InMemoryRepo extends Repository[User] with Serializable:
  private val users = mutable.HashMap[Int, User]()

  override def findById(id: Int): Option[User] =
    users.get(id)

  override def save(user: User): Unit =
    users.put(user.id, user)
    println(s"Saved user ${user.id}")

class UserService(repo: Repository[User]):
  def createUser(name: String, email: String): User =
    val user = User(1, name, email)
    repo.save(user)
    user

  def getUser(id: Int): Option[User] =
    repo.findById(id)

object UserService:
  def apply(repo: Repository[User]): UserService = new UserService(repo)

enum Color:
  case Red, Green, Blue
